src/components/post/post.service.ts

```diff
@@ -0,0 +1,313 @@
+              createdAt: DateTime.local(),
+            }),
+            node('post'),
+          ])
+          .return('post.id as id')
+          .first();
+      }
+      // connect the post to the engagement, if applicable
+      else if (engagementId) {
+        await this.db
+          .query()
+          .match([
+            [node('engagement', 'Engagement', { id: engagementId })],
+            [node('post', 'Post', { id: postId })],
+          ])
+          .create([
+            node('engagement'),
+            relation('out', '', 'engagement', {
+              active: true,
+              createdAt: DateTime.local(),
+            }),
+            node('post'),
+          ])
+          .return('post.id as id')
+          .first();
+      }
+      // connect the post to a parent post, if applicable
+      else if (parentPostId) {
+        await this.db
+          .query()
+          .match([
+            [node('parentPost', 'Post', { id: parentPostId })],
+            [node('post', 'Post', { id: postId })],
+          ])
+          .create([
+            node('parentPost'),
+            relation('out', '', 'parentPost', {
+              active: true,
+              createdAt: DateTime.local(),
+            }),
+            node('post'),
+          ])
+          .return('post.id as id')
+          .first();
+      }
+
+      // FIXME: This is being refactored - leaving it commented out per Michael's instructions for now
+      // await this.authorizationService.processNewBaseNode(
+      //   new DbPost(),
+      //   postId,
+      //   session.userId
+      // );
+
+      return await this.readOne(postId, session);
+    } catch (exception) {
+      this.logger.warning('Failed to create post', {
+        exception,
+      });
+
+      throw new ServerException('Failed to create post', exception);
+    }
+  }
+
+  async readOne(postId: string, session: Session): Promise<Post> {
+    const query = this.db
+      .query()
+      .call(matchRequestingUser, session)
+      .match([node('node', 'Post', { id: postId })])
+      .call(matchPropList)
+      .match([node('node'), relation('out', '', 'user'), node('user', 'User')])
+      .return('node, propList, user.id as userId')
+      .asResult<
+        StandardReadResult<DbPropsOfDto<Post>> & {
+          userId: string;
+        }
+      >();
+
+    const result = await query.first();
+
+    if (!result) {
+      throw new NotFoundException('Could not find post', 'post.id');
+    }
+
+    const props = parsePropList(result.propList);
+    const securedProps = await this.authorizationService.secureProperties(
+      Post,
+      result.propList,
+      session
+    );
+
+    return {
+      ...parseBaseNodeProperties(result.node),
+      ...securedProps,
+      body: {
+        value: props.body,
+        canRead: true,
+        canEdit: true,
+      },
+      user: {
+        ...securedProps.user,
+        value: await this.userService.readOne(result.userId, session),
+      },
```

Comment:
If we cannot grab the user in the same db query then we should defer that to another resolver. It allows this query to be skipped entirely if `user` is not asked for. 
See these examples:
https://github.com/SeedCompany/cord-api-v3/blob/bcfef17a3c44066d8f5f247825f9e5035f18621e/src/components/engagement/dto/engagement.dto.ts#L45-L45
https://github.com/SeedCompany/cord-api-v3/blob/bcfef17a3c44066d8f5f247825f9e5035f18621e/src/components/engagement/engagement.resolver.ts#L65-L76